app/javascript/react/components/capeform/SearchBar.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from 'react';
+import { Redirect } from 'react-router-dom';
+import ResultTile from './ResultTile';
+
+const SearchBar = () => {
+  const [query, setQuery] = useState('');
+  const [results, setResults] = useState([]);
+  const [redirect, setRedirect] = useState(false);
+  const [showPagePathId, setShowPagePathId] = useState(0);
+
+  let mappedSuperHeroes;
+  let externalHero;
+  const key = '2937630366248635';
+  const superheroesUrl = `https://cors-anywhere.herokuapp.com/https://www.superheroapi.com/api.php/${key}/search/${query}`;
+
+  const handleChange = event => {
+    console.log(process.env.REACT_APP_API_KEY)
+
+    setQuery(event.target.value);
+  };
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    findHero();
+  };
+
+  function findHero() {
+    fetch(superheroesUrl)
```

Comment:
there is no point in separating handleSubmit and findHero. Do this all in handlesubmit